connectors/sources/github.py

```diff
@@ -1834,44 +1846,54 @@ async def _fetch_last_commit_timestamp(self, repo_name, path):
         )
         return commit["commit"]["committer"]["date"]
 
-    async def _fetch_files(self, repo_name, default_branch):
+    async def _fetch_files(self, repo_name, path_branch, path=False):
```

Comment:
Can we break this method into two: `fetch_files` with old signature/logic and `fetch_path` with new logic to fetch filtering by path?